homeassistant/components/simplisafe/camera.py

```diff
@@ -0,0 +1,243 @@
+        simplisafe: SimpliSafe,
+        system: SystemV3,
+        camera: DeviceV3,
+    ) -> None:
+        """Initialize."""
+        SimpliSafeEntity.__init__(
+            self,
+            simplisafe,
+            system,
+            device=camera,
+            additional_websocket_events=[EVENT_CAMERA_MOTION_DETECTED],
+        )
+        Camera.__init__(self)
+
+        self._hass = hass
+        self._register_services()
+
+    @property
+    def name(self) -> str | None:
+        """Return a good name for this camera."""
+        return self._device.name
+
+    @callback
+    def async_unload(self) -> None:
+        """Release resources."""
+        for service in SERVICES:
+            self._hass.services.async_remove(DOMAIN, service)
+
+    async def async_camera_motion_image(
+        self, width: int | None = None, height: int | None = None
+    ) -> bytes | None:
+        """Return bytes of camera image."""
+        if self._attr_cached_image is not None:
+            return self._attr_cached_image
+
+        if self._attr_image_url is None:
+            return None
+
+        self._attr_cached_image = await self._simplisafe.async_media_request(
+            self._attr_image_url.replace("{&width}", "&width=" + str(width))
+        )
+        return self._attr_cached_image
+
+    def _register_services(self) -> None:
+        """Register the Outdoor Camera hass service calls."""
+
+        def _write_image(to_file: str, image_data: bytes) -> None:
+            """Executor helper to write image."""
+            os.makedirs(os.path.dirname(to_file), exist_ok=True)
+            with open(to_file, "wb") as img_file:
+                img_file.write(image_data)
+
+        async def save_image_handler(call: ServiceCall) -> None:
+            if self._attr_image_url is None:
+                return
+
+            width = call.data.get(ATTR_WIDTH)
+            filename: Template = call.data.get(ATTR_FILENAME)  # type: ignore[assignment]
+            filename.hass = self._hass
+            snapshot_file = filename.async_render(variables={ATTR_ENTITY_ID: self})
+
+            try:
+                snapshot = await self._simplisafe.async_media_request(
+                    self._attr_image_url.replace("{&width}", "&width=" + str(width))
+                )
+            except SimplipyError as err:
+                raise HomeAssistantError(
+                    f'Error fetching motion media "{self._system.system_id}": {err}'
+                ) from err
+
+            try:
+                await self._hass.async_add_executor_job(
+                    _write_image, snapshot_file, snapshot
+                )
+            except OSError as err:
+                LOGGER.error("Can't write image to file: %s", err)
+
+        async def save_clip_handler(call: ServiceCall) -> None:
+            if self._attr_clip_url is None:
+                return
+
+            filename: Template = call.data.get(ATTR_FILENAME)  # type: ignore[assignment]
```

Comment:
Why are we disabling `mypy` here?

Reply:
Because I get: error: Incompatible types in assignment (expression has type "Any | None", variable has type "Template")  [assignment]